R/epiparameter.R

```diff
@@ -0,0 +1,996 @@
+#' transmitted), etc. It is assumed that the disease is not
+#' vector-borne and that the distribution is intrinsic (e.g. not an extrinsic
+#' delay distribution such as extrinsic incubation period) unless
+#' `transmission_mode = "vector_borne"` is contained in the metadata. Use
+#' [create_metadata()] to create metadata.
+#' @param method_assess A list of methodological aspects used when fitting
+#' the distribution, use [create_method_assess()] to create method
+#' assessment.
+#' @param notes A `character` string with any additional information about the
+#' data, inference method or disease.
+#' @param ... [dots] Extra arguments to be passed to internal functions.
+#'
+#' This is most commonly used to pass arguments to [distcrete::distcrete()]
+#' that construct the discretised distribution S3 object. To see which
+#' arguments can be adjusted for discretised distributions see
+#' [distcrete::distcrete()].
+#'
+#' @return An `<epiparameter>` object.
+#' @export
+#'
+#' @examples
+#' # minimal input required for `epiparameter`
+#' ebola_incubation <- epiparameter(
+#'   disease = "ebola",
+#'   epi_name = "incubation_period",
+#'   prob_distribution = create_prob_distribution(
+#'     prob_distribution = "gamma",
+#'     prob_distribution_params = c(shape = 1, scale = 1)
+#'   )
+#' )
+#'
+#' # minimal input required for discrete `epiparameter`
+#' ebola_incubation <- epiparameter(
+#'   disease = "ebola",
+#'   epi_name = "incubation_period",
+#'   prob_distribution = create_prob_distribution(
+#'     prob_distribution = "gamma",
+#'     prob_distribution_params = c(shape = 1, scale = 1),
+#'     discretise = TRUE
+#'   )
+#' )
+#'
+#' # example with more fields filled in
+#' ebola_incubation <- epiparameter(
+#'   disease = "ebola",
+#'   pathogen = "ebola_virus",
+#'   epi_name = "incubation",
+#'   prob_distribution = create_prob_distribution(
+#'     prob_distribution = "gamma",
+#'     prob_distribution_params = c(shape = 1, scale = 1),
+#'     discretise = FALSE,
+#'     truncation = NA
+#'   ),
+#'   uncertainty = list(
+#'     shape = create_uncertainty(),
+#'     scale = create_uncertainty()
+#'   ),
+#'   summary_stats = create_summary_stats(mean = 2, sd = 1),
+#'   citation = create_citation(
+#'     author = person(given = "John", family = "Smith"),
+#'     year = 2002,
+#'     title = "COVID-19 incubation period",
+#'     journal = "Epi Journal",
+#'     doi = "10.19832/j.1366-9516.2012.09147.x"
+#'   ),
+#'   metadata = create_metadata(
+#'     units = "days",
+#'     sample_size = 10,
+#'     region = "UK",
+#'     transmission_mode = "natural_human_to_human",
+#'     inference_method = "MLE"
+#'   ),
+#'   method_assess = create_method_assess(
+#'     censored = TRUE
+#'   ),
+#'   notes = "No notes"
+#' )
+epiparameter <- function(disease,
+                         pathogen = NA_character_,
+                         epi_name,
+                         prob_distribution = create_prob_distribution(
```

Comment:
I personally prefer to use the helper functions as argument defaults as I think it helps navigate users towards these functions which they may otherwise not know exist. I agree in this case it is an unnecessary abstraction, but for consistency with the rest of the package I will leave as is.